Sources/JWTProvider/PayloadAuthenticationMiddleware.swift

```diff
@@ -64,22 +64,24 @@ public final class PayloadAuthenticationMiddleware<U: PayloadAuthenticatable>: M
 
         let filteredSigners = try self.signers(for: jwt)
 
-        // Try to use all the signers until one matches
-        var verified = false
+        // Default Error
+        var error: Swift.Error? = JWTError.signatureVerificationFailed
 
+        // Try to use all the signers until one matches
         for signer in filteredSigners {
 
             do {
-                _ = try req.jwt(verifyUsing: signer, and: claims)
-                verified = true
+                _ = try req.jwt(verifyUsing: signer, and: self.claims)
+                error = nil
                 break
-            } catch {
+            } catch let jwtError {
+                error = jwtError
                 continue
             }
         }
 
-        guard verified else {
-            throw JWTProviderError.noVerifiedJWT
```

Comment:
this is the only case causing that method to return an array with more than one element:

```
guard let kid = jwt.keyIdentifier else {
            // The token doesn't include a kid, try to verify using all the signers
            return Array(self.signers.values)
        }
```

We could remove that logic, but then jwt tokens without `kid` won't be verified anymore, so it won't be backwards compatible. I am not sure we want to enforce people to use `kid`, although it's good practice.